### PR TITLE
Graduate SeccompDefault feature to beta

### PR DESCRIPTION
**Pull request by saschagrunert**
We now update the documentation to reflect the current state of the feature.

Refers to: https://github.com/kubernetes/enhancements/issues/2413
k/k PR: https://github.com/kubernetes/kubernetes/pull/110805

### CODE REVIEW

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -178,7 +178,7 @@ different Kubernetes components.
 | `RemainingItemCount` | `true` | Beta | 1.16 | |
 | `RotateKubeletServerCertificate` | `false` | Alpha | 1.7 | 1.11 |
 | `RotateKubeletServerCertificate` | `true` | Beta | 1.12 | |
-| `SeccompDefault` | `false` | Alpha | 1.22 | |
+| `SeccompDefault` | `true` | Beta | 1.25 | |
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:181]: You meant this, I think:
```suggestion
| `SeccompDefault` | `false` | Alpha | 1.22 | 1.24 |
| `SeccompDefault` | `true` | Beta | 1.25 | |
```

File: content/en/docs/tutorials/security/seccomp.md [hunk 2]
```diff
@@ -199,10 +199,10 @@ test workload execution before rolling the change out cluster-wide.
 More detailed information about a possible upgrade and downgrade strategy can be
 found in the [related Kubernetes Enhancement Proposal (KEP)](https://github.com/kubernetes/enhancements/tree/a70cc18/keps/sig-node/2413-seccomp-by-default#upgrade--downgrade-strategy).
 
-Since the feature is in alpha state it is disabled per default. To enable it,
-pass the flags `--feature-gates=SeccompDefault=true --seccomp-default` to the
-`kubelet` CLI or enable it via the [kubelet configuration
-file](/docs/tasks/administer-cluster/kubelet-config-file/). To enable the
+`SeccompDefault` is in beta state, which means that the corresponding feature
+gate is `true` by default. To additionally enable the feature, pass the
+flag `--seccomp-default` to the `kubelet` CLI or enable it via the [kubelet
```
Comment (sftim) [inline content/en/docs/tutorials/security/seccomp.md:204]:
```suggestion
Seccomp defaulting for Pods is a beta feature in Kubernetes {{< skew currentVersion >}}, 
and the corresponding `SeccompDefault` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
is enabled by default. However, you still need to enable this defaulting for each node where
you would like to use it.

If you are running Kubernetes {{< skew currentVersion >}} cluster and want to enable Seccomp
defaulting, either run the kubelet with the `--seccomp-default` command line flag, or enable
Seccomp defaulting through the
[kubelet
```

File: content/en/docs/tutorials/security/seccomp.md [hunk 3]
```diff
@@ -199,10 +199,10 @@ test workload execution before rolling the change out cluster-wide.
 More detailed information about a possible upgrade and downgrade strategy can be
 found in the [related Kubernetes Enhancement Proposal (KEP)](https://github.com/kubernetes/enhancements/tree/a70cc18/keps/sig-node/2413-seccomp-by-default#upgrade--downgrade-strategy).
```
Comment (sftim) [inline content/en/docs/tutorials/security/seccomp.md:200]:
```suggestion
You can find more detailed information about a possible upgrade and downgrade strategy
in the related Kubernetes Enhancement Proposal (KEP):
[Enable seccomp by default](https://github.com/kubernetes/enhancements/tree/9a124fd29d1f9ddf2ff455c49a630e3181992c25/keps/sig-node/2413-seccomp-by-default#upgrade--downgrade-strategy).
```
Notice that I bumped to commit https://github.com/kubernetes/enhancements/commit/9a124fd29d1f9ddf2ff455c49a630e3181992c25

File: content/en/docs/tutorials/security/seccomp.md [hunk 4]
```diff
@@ -159,7 +159,7 @@ running within kind.
 
 ## Enable the use of `RuntimeDefault` as the default seccomp profile for all workloads
 
-{{< feature-state state="alpha" for_k8s_version="v1.22" >}}
+{{< feature-state state="beta" for_k8s_version="v1.25" >}}
 
 `SeccompDefault` is an optional kubelet
 [feature gate](/docs/reference/command-line-tools-reference/feature-gates) as
```
Comment (sftim) [inline content/en/docs/tutorials/security/seccomp.md:165]:
```suggestion
To use seccomp profile defaulting, you must run the kubelet with the `SeccompDefault`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) enabled
(this is the default). You must also explicitly enable the defaulting behavior for each
node where you want to use this.
```
**(and then revise the next line).**

**Review by sftim — COMMENTED** [review]
Some changes needed, I'm afraid.

**Comment by saschagrunert** [comment]
> Some changes needed, I'm afraid.

Thanks for the fast review, I incorporated the requested changes. :+1:

File: content/en/docs/tutorials/security/seccomp.md [hunk 5]
```diff
@@ -196,13 +198,20 @@ If you were introducing this feature into production-like cluster, the Kubernete
...
-Since the feature is in alpha state it is disabled per default. To enable it,
-pass the flags `--feature-gates=SeccompDefault=true --seccomp-default` to the
-`kubelet` CLI or enable it via the [kubelet configuration
-file](/docs/tasks/administer-cluster/kubelet-config-file/). To enable the
+Seccomp defaulting for Pods is a beta feature in Kubernetes {{< skew currentVersion >}}, 
+and the corresponding `SeccompDefault` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+is enabled by default. However, you still need to enable this defaulting for each node where
+you would like to use it.
+
+If you are running Kubernetes {{< skew currentVersion >}} cluster and want to enable Seccomp
```
Comment (sftim) [inline content/en/docs/tutorials/security/seccomp.md:210]: Oops.
```suggestion
If you are running Kubernetes a {{< skew currentVersion >}} cluster and want to enable Seccomp
```

File: content/en/docs/tutorials/security/seccomp.md [hunk 6]
```diff
@@ -159,10 +159,12 @@ running within kind.
...
-{{< feature-state state="alpha" for_k8s_version="v1.22" >}}
+{{< feature-state state="beta" for_k8s_version="v1.25" >}}
 
-`SeccompDefault` is an optional kubelet
-[feature gate](/docs/reference/command-line-tools-reference/feature-gates) as
+To use seccomp profile defaulting, you must run the kubelet with the `SeccompDefault`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) enabled
+(this is the default). You must also explicitly enable the defaulting behavior for each
+node where you want to use this.
 well as corresponding `--seccomp-default`
```
Comment (mtardy) [inline content/en/docs/tutorials/security/seccomp.md:168]:
```suggestion
node where you want to use this with the corresponding `--seccomp-default`
```

Comment (saschagrunert) [reply content/en/docs/tutorials/security/seccomp.md]: Good catch, thank you!

**Review by pjbgf — APPROVED** [review]
@saschagrunert thanks for the continuous work on seccomp. :bow: 

/lgtm

File: content/en/docs/tutorials/security/seccomp.md [hunk 7]
```diff
@@ -112,7 +112,7 @@ See [Nodes](https://kind.sigs.k8s.io/docs/user/configuration/#nodes) within the
 kind documentation about configuration for more details on this.
 This tutorial assumes you are using Kubernetes {{< param "version" >}}.
 
-As an alpha feature, you can configure Kubernetes to use the profile that the
+As an beta feature, you can configure Kubernetes to use the profile that the
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply content/en/docs/tutorials/security/seccomp.md]: Thanks, fixed!

**Comment by saschagrunert** [comment]
k/k PR got merged, can we also merge this one now?

File: content/en/docs/tutorials/security/seccomp.md [hunk 8]
```diff
@@ -196,13 +197,20 @@ If you were introducing this feature into production-like cluster, the Kubernete
...
-found in the [related Kubernetes Enhancement Proposal (KEP)](https://github.com/kubernetes/enhancements/tree/a70cc18/keps/sig-node/2413-seccomp-by-default#upgrade--downgrade-strategy).
+You can find more detailed information about a possible upgrade and downgrade strategy
+in the related Kubernetes Enhancement Proposal (KEP):
+[Enable seccomp by default](https://github.com/kubernetes/enhancements/tree/9a124fd29d1f9ddf2ff455c49a630e3181992c25/keps/sig-node/2413-seccomp-by-default#upgrade--downgrade-strategy).
 
-Since the feature is in alpha state it is disabled per default. To enable it,
-pass the flags `--feature-gates=SeccompDefault=true --seccomp-default` to the
-`kubelet` CLI or enable it via the [kubelet configuration
-file](/docs/tasks/administer-cluster/kubelet-config-file/). To enable the
+Seccomp defaulting for Pods is a beta feature in Kubernetes {{< skew currentVersion >}},
```
Comment (kbhawkey) [inline content/en/docs/tutorials/security/seccomp.md:204]: Question: Is the feature called 'Seccomp defaulting' or 'Seccomp'?
I'm not sure I understand the use of the word, 'defaulting'. Which or what is 'defaulting'?
`However, you still need to enable this defaulting for each node where
you would like to use it.`

**Comment by kbhawkey** [comment]
Thanks @saschagrunert 
/approve

File: content/en/docs/tutorials/security/seccomp.md (hunk 8 above)
Comment (saschagrunert) [reply content/en/docs/tutorials/security/seccomp.md:204]: Right, fixing it in https://github.com/kubernetes/website/pull/35121

**Pull request merged**